### DISCUSSION TITLE
Estimate when the AWS Kotlin SDK will be out of beta?

### DISCUSSION THREAD

**Discussion by scmorse** [aws/aws-sdk-kotlin#744] [General]
I checked out the roadmap, but didn't see anything that communicated a release timeline. I'm looking to know when this AWS Kotlin SDK will be cleared for production use? JVM/backend is the platform I'm targeting, though broader timelines could be useful for others as well.

**Comment by aajtodd** [comment]
@scmorse 

Thanks for the question. We won't mark the SDK as production ready until we are ready to do a GA release. That isn't a useful answer in determining "production ready" though. The biggest difference between where we are at now and an eventual GA is (1) backwards compatibility and (2) feature completeness. If we have the features you need already and you are willing to deal with _potentially_ breaking changes when upgrading between now and when we GA then I'd encourage you to try out the SDK and provide feedback.

Bugs can and will happen whether it is marked production ready or not and they would be fixed appropriately either way.

**Reply by scmorse** [reply]
Thanks for the reply @aajtodd! It does look like the SDK has good support for DynamoDB at this point, which is what my team would primarily be interested in, but we don't have any non-production projects to try out the SDK on currently. Do you think the DynamoDB portion of the AWS Kotlin SDK is essentially GA-ready? 

It's hard to convince a team to use a library that is clearly marked as not-production-ready, though, so I'm thinking we'll have to continue to use the Java SDK, but I look forward to your eventual GA release! Thanks!

**Reply by efemoney** [reply]
Fwiw, we use the Kotlin SDK in prod w/o **a single issue** for more than a year now in a user-facing service (primarily DynamoDB however).

My personal opinion? ... go for it!

**Comment by lorrainefox** [comment]
It's August 2023 and I'm wondering if you have any estimates as to when GA will be?  I'm working on a kotlin small project that will use DynamoDB, simple use cases only, but would love to use the kotlin SDK.  If GA is near, I may be able to convince my management.

**Reply by aajtodd** [reply]
Thanks for your interest in the SDK. The best answer I can give is unfortunately the one already given [above](https://github.com/awslabs/aws-sdk-kotlin/discussions/744#discussioncomment-4079953). The primary difference between now and GA is API stabilization which is not the same thing as production readiness. We encourage you to try the SDK and see if it fits your needs (and provide feedback good or bad!).
